### PR TITLE
fix clippy lints

### PR DESCRIPTION
**Pull request by kczimm**
Fixed clippy warnings. I'm not sure if these changes are of general interest or if they are viewed as nitpicking. Would love the feedback.

### CODE REVIEW

File: pgml-extension/src/orm/snapshot.rs [hunk 1]
```diff
@@ -141,10 +141,10 @@ impl Column {
     }
 
     fn nominal_type(pg_type: &str) -> bool {
-        match pg_type {
-            "bpchar" | "text" | "varchar" | "bpchar[]" | "text[]" | "varchar[]" => true,
-            _ => false,
-        }
+        matches!(
```
Comment (levkk) [inline pgml-extension/src/orm/snapshot.rs:144]: This is cool

File: pgml-extension/src/orm/snapshot.rs [hunk 2]
```diff
@@ -283,16 +279,18 @@ impl Column {
 
         let mut modes = Vec::new();
         statistics.distinct = 0; // necessary reset before array columns clobber
-        for &value in &data {
+        for (i, &value) in data.iter().enumerate() {
```
Comment (levkk) [inline pgml-extension/src/orm/snapshot.rs:282]: Why would it suggest that? `i` is not used, is it?

**Comment by levkk** [comment]
Cool! Can't argue with clippy, this is definitely better than it was.

File: pgml-extension/src/orm/snapshot.rs (hunk 2 above)
Comment (kczimm) [reply pgml-extension/src/orm/snapshot.rs:282]: It's used on line 307 I believe when calculating `v`. Previously it was used as an explicit counter outside of the for loop. https://rust-lang.github.io/rust-clippy/master/index.html#/explicit_counter_loop

Comment (levkk) [reply pgml-extension/src/orm/snapshot.rs:282]: That's funny, wonder how we missed that !

**Comment by montanalow** [comment]
At some point we should add clippy back to CI, as it's nice for everyone to have an idiomatic standard.

**Comment by kczimm** [comment]
Should that hypothetically be a separate PR or should I try to add it in here?

**Comment by montanalow** [comment]
I don't think it's critically urgent, or should block this PR.

**Pull request merged**